lib/assets/connection_cell/main.js

```diff
@@ -4,12 +4,301 @@ export function init(ctx, info) {
+        :grow
+      />
+      <BaseInput
+        name="database"
+        label="Database"
+        type="text"
+        placeholder="Database"
+        v-model="fields.database"
+        inputClass="input"
+        :grow
+      />
+    </div>
+    <div class="row">
+      <BaseInput
+        name="username"
+        label="User"
+        type="text"
+        placeholder="User"
+        v-model="fields.username"
+        inputClass="input"
+        :grow
+      />
+      <BaseInput
+        name="password"
+        label="Password"
+        type="password"
+        placeholder="Password"
+        v-model="fields.password"
+        inputClass="input"
+        :grow
+      />
+    </div>
+    `
+  };
+
+  const BigQueryForm = {
+    name: "BigQueryForm",
+
+    components: {
+      BaseInput: BaseInput
+    },
+
+    props: {
+      fields: {
+        type: Object,
+        default: {}
+      },
+    },
+
+    methods: {
+      credentialsChange(_) {
+        this.updateCredentials(this.$refs.credentials.files);
+      },
+
+      credentialsClick(_) {
+        this.$refs.credentials.click();
+      },
+
+      dragOver(event) {
+        event.preventDefault();
+      },
+
+      dragLeave(_) { },
+
+      drop(event) {
+        event.preventDefault();
+        this.updateCredentials(event.dataTransfer.files);
+      },
+
+      updateCredentials(fileList) {
+        const file = fileList[0];
+
+        if (file && file.type === "application/json") {
+          const reader = new FileReader();
+
+          reader.onload = (res) => {
+            const value = JSON.parse(res.target.result);
+            ctx.pushEvent("update_field", { field: "credentials", value });
+          };
+
+          reader.readAsText(file);
+        }
+      }
+    },
+
+    template: `
+    <div class="row mixed-row">
+      <BaseInput
+        name="project_id"
+        label="Project ID"
+        type="text"
+        placeholder="Project ID"
+        v-model="fields.project_id"
+        inputClass="input"
+        :grow
+      />
+      <BaseInput
+        name="default_dataset_id"
+        label="Default Dataset ID (Optional)"
+        type="text"
+        placeholder="Default Dataset ID (Optional)"
+        v-model="fields.default_dataset_id"
+        inputClass="input"
+        :grow
+      />
+    </div>
+    <div class="row">
+      <div class="draggable" @dragover="dragOver" @dragleave="dragLeave" @drop="drop" @click="credentialsClick">
+        <label for="credentials">
+          Drag your credentials JSON file here<br/>
+          or click here to select your file.
+        </label>
+        <input type="file" ref="credentials" @change="credentialsChange" />
+      </div>
+    </div>
+    <small class="help-box">
+      To generate the Google BigQuery Credentials JSON file,
+      please take a look on <a href="https://cloud.google.com/iam/docs/creating-managing-service-account-keys" target="_top">this link</a>.
```

Comment:
Only `_blank` will work and that's likely a reasonable behaviour anyway :)

```suggestion
      please take a look on <a href="https://cloud.google.com/iam/docs/creating-managing-service-account-keys" target="_blank">this link</a>.
```